build-tools/packages/build-tools/src/repoPolicyCheck/handlers/npmPackages.ts

```diff
@@ -1416,6 +1416,108 @@ export const handlers: Handler[] = [
 				}
 			});
 
+			return result;
+		},
+	},
+	{
+		// Verify that non-private packages are configured correctly for API-Extractor by checking for the following:
+		// - dev dependency on `@microsoft/api-extractor`
+		// - `build:docs` script
+		// - `check:release-tags` script
+		name: "api-extractor-public-packages",
+		match,
+		handler: async (file) => {
+			let json: PackageJson;
+			try {
+				json = JSON.parse(readFile(file));
+			} catch (err) {
+				return "Error parsing JSON file: " + file;
+			}
+
+			const packageName = json.name;
+
+			if (json.private || packageName.startsWith("@fluid-private")) {
```

Comment:
This seems like a sensible policy to me, but I'm interested in thoughts from reviewers. If the package is repo-private (i.e. contains `"private": true`, OR if the package is one of our `@fluid-private` packages that only get published to our internal feed.

Reply:
Seems reasonable to me. Already covers the examples which should be `private: true`, with the known exception of table-document. We still have packages labeled @fluid-internal that will become @fluid-private but aren't yet, AFAIK. Will those fail the new handler?

Reply:
The relevant `@fluid-internal` packages that have not been migrated to `@fluid-private` yet would already have the `private` field set, so we should be good there. Ones that get published should likely have all of these api-extractor invariants met.